app/ide-desktop/client/src/index.ts

```diff
@@ -151,6 +153,33 @@ class App {
     }
   }
 
+  /** Background tasks scheduled on the application startup. */
+  async onStart() {
+    const userData = electron.app.getPath('userData')
+    const versionInfoPath = pathModule.join(userData, 'version_info.json')
```

Comment:
Here we should respect the user preference for the `RootDirectory`

Reply:
This should be the path to the settings folder of the app which I think is correct.

Reply:
`userData` represents internal Electron directory with user data

Reply:
@4e6 I think we definitely need to test this PR with the custom root folder, you can change it here:
![CleanShot 2025-03-03 at 20 09 31@2x](https://github.com/user-attachments/assets/4d1d91e9-59fb-4e74-8842-0f316128ca2d)

Reply:
@4e6 on the second though my comment seems related to this line: https://github.com/enso-org/enso/pull/12299/files#diff-1dde77591926f101b841920608691aedee7666d5b7a9dbc2eef6b3a3a61daed5R472